crates/node/rpc/src/jsonrpsee.rs

```diff
@@ -0,0 +1,173 @@
+//! The Optimism RPC API using `jsonrpsee`
+
+use alloc::{boxed::Box, string::String, vec::Vec};
+use core::net::IpAddr;
+
+use alloy_eips::BlockNumberOrTag;
+use jsonrpsee::{core::RpcResult, proc_macros::rpc};
+use kona_genesis::RollupConfig;
+use kona_interop::{ExecutingMessage, SafetyLevel};
+use kona_protocol::SyncStatus;
+
+#[cfg_attr(all(target_arch = "wasm32", target_os = "unknown"), allow(unused_imports))]
+use getrandom as _; // required for compiling wasm32-unknown-unknown
+
+use crate::{
+    OutputResponse, PeerDump, PeerInfo, PeerStats, ProtocolVersion, SafeHeadResponse,
+    SuperchainSignal,
+};
+
+// Re-export apis defined in upstream `op-alloy-rpc-jsonrpsee`
+pub use op_alloy_rpc_jsonrpsee::traits::{MinerApiExtServer, OpAdminApiServer};
+
+#[cfg(feature = "client")]
+pub use op_alloy_rpc_jsonrpsee::traits::{MinerApiExtClient, OpAdminApiClient};
+
+/// Optimism specified rpc interface.
+///
+/// https://docs.optimism.io/builders/node-operators/json-rpc
+/// https://github.com/ethereum-optimism/optimism/blob/8dd17a7b114a7c25505cd2e15ce4e3d0f7e3f7c1/op-node/node/api.go#L114
+#[cfg_attr(not(feature = "client"), rpc(server, namespace = "optimism"))]
+#[cfg_attr(feature = "client", rpc(server, client, namespace = "optimism"))]
+pub trait RollupNode {
```

Comment:
Small nit, can we rename this to `RollupNodeApi` or something to the effect of that?

We've a struct named `RollupNode` for the core service over in `bin/node`

Reply:
out of scope see